core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -134,14 +136,14 @@ public long targetSplitSize() {
               .withRecordCount(1)
               .withFormat(FileFormat.AVRO)
               .build();
-          return new ManifestListReadTask(ops.io(), schema(), table().spec(), new BaseFileScanTask(
+          return new ManifestListReadTask(ops.io(), schema(), specs, new BaseFileScanTask(
               manifestListAsDataFile, null,
               schemaString, specString, residuals));
         } else {
           return StaticDataTask.of(
               ops.io().newInputFile(ops.current().metadataFileLocation()),
               MANIFEST_FILE_SCHEMA, schema(), snap.allManifests(),
-              manifest -> ManifestsTable.manifestFileToRow(table().spec(), manifest)
+              manifest -> ManifestsTable.manifestFileToRow(specs.get(manifest.partitionSpecId()), manifest)
```

Comment:
Is this change needed, shouldn't we be using the common spec from table built in Partitioning?

Reply:
`Partitioning.partitionType` builds a common partition type, here we need a `PartitionSpec` argument.